code/modules/mob/living/silicon/laws.dm

```diff
@@ -1,7 +1,93 @@
+#define MAX_INTEGRITY 100
+#define MIN_INTEGRITY -100
+#define MODE_FACTORY "factory-default"
+#define MODE_FIRMWARE "load-firmware-laws"
+#define MODE_LISP "lawchange-induced-synthetic-psychosis"
+#define MODE_REGULAR "regular"
+#define THRESHOLD_LISP -50
+#define THRESHOLD_FACTORY 80
+
 /mob/living/silicon
 	var/datum/ai_laws/laws = null
 	var/list/additional_law_channels = list("State" = "")
-	var/lawcooldown = FALSE
+	var/integrity = MAX_INTEGRITY
+	var/law_mode = MODE_REGULAR
+
+/mob/living/silicon/proc/adjust_integrity(amount=0)
+	var/old_integrity = clamp(integrity, MIN_INTEGRITY, MAX_INTEGRITY)
+	var/new_integrity = clamp((integrity + amount), MIN_INTEGRITY, MAX_INTEGRITY)
+
+	// if we're in regular mode
+	if(law_mode == MODE_REGULAR)
```

Comment:
Switch would be good here, instead of the if else's:
```js
switch(law_mode)
    if(MODE_REGULAR)
        ...
    if(MODE_FACTORY)
        ...
    // etc.
```